esp-hal/src/gpio/mod.rs

```diff
@@ -1493,6 +1727,8 @@ where
 }
 
 /// Flexible pin driver.
+///
+/// This driver allows changing the pin mode between input and output.
```

Comment:
Flex is intentionally left undocumented. I don't think I can come up with a good but simple real-world example, and I think it's mostly targeting advanced users anyway. Thirdly, I'm not sure about its public API at this point. While other drivers may be missing functions, Flex I think deserves a bit of a makeover.